content/resources/operator/operator.md

```diff
@@ -0,0 +1,66 @@
++++
+fragment = "content"
+#disabled = true
+date = "2020-12-07"
+weight = 100
+# title = "Creating and modifying an operator in Kubernets"
++++
+
+
+
+## Introduction
+
+CrownLabs uses the Kubernetes [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ "operator pattern") to automate complex tasks such as the configuration and provisioning of virtual environments.
+With this paradigm we can create both new _custom resources_, a set of objects that model a specific part of system (e.g., the _student_ with its own set of _attributes_), and the associated _software logic_ (e.g., the code required to _create_ a new student, _update_ or _delete_ this object, and more).
+As a consequence, the implementation of additional features in CrownLabs implies dealing with operators, either modifying an existing one or creating a new one.
+
+In CrownLabs, most operators use the [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) framework to simplify the task of generating [CRDs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/ "CRDs"), Controllers (where the operator\'s logic resides) and the various Kubernetes manifests needed.
+
+This page aims at providing a set of useful tips and references to start learning and practicing with operators.
+
+This page is split in two sections, depending whether you want to create a _new_ operator from scratch or you want to modify an _existing_ one; for instance, it is very common that a new feature in CrownLabs may require modifications to existing operators instead of asking for a brand new operator.
+
+
+## Prerequisites
+Regardless you want to create a _new_ operator or modify an _existing_ one,  here there are the software prerequisites.
+
+1. go version v1.13+
+2. docker version 17.03+
+3. kubectl version v1.11.3+
+4. Access to a Kubernetes v1.11.3+ cluster ([kind](https://kind.sigs.k8s.io/ "kind") or [minikube](https://minikube.sigs.k8s.io/docs/ "minikube") are good enough)
+5. kubebuilder ([installation](https://book.kubebuilder.io/quick-start.html#installation "installation"))
+
+
+## Creating an operator
+
+Useful information to get the kubebuilder project started:
+- Use `polito.it` as domain when doing `kubebuilder init` (or just `example.com`  if you\'re just testing stuff)
```

Comment:
In the written text we should avoid shortening verbs. E.g. you're ==> you are; don't ==> do not, etc.